projects/netgrif-components-core/src/lib/view/task-view/abstract-single-task-view.component.ts

```diff
@@ -17,22 +18,32 @@ export abstract class AbstractSingleTaskViewComponent extends AbstractViewWithHe
 
     @Input() initiallyExpanded: boolean = true;
     @Input() preventCollapse: boolean = true;
+
+    @Output() noTaskPresent: EventEmitter<void>;
     public taskPanelData: ReplaySubject<TaskPanelData>;
     public loading$: Observable<boolean>;
     private transitionId: string;
     private subRoute: Subscription | undefined;
     protected subPanelData: Subscription | undefined;
+    protected subLoading: Subscription | undefined;
 
     protected constructor(protected taskViewService: TaskViewService,
-                          activatedRoute: ActivatedRoute) {
+                          activatedRoute: ActivatedRoute, async: AsyncPipe) {
         super(taskViewService, activatedRoute);
+        this.noTaskPresent = new EventEmitter<void>();
         this.taskPanelData = new ReplaySubject<TaskPanelData>(1);
         this.subRoute = this._activatedRoute.paramMap.subscribe(paramMap => {
             if (!!(paramMap?.['params']?.[TaskConst.TRANSITION_ID])) {
                 this.transitionId = paramMap['params'][TaskConst.TRANSITION_ID];
                 this.subPanelData = this.taskViewService.tasks$.subscribe(tasks =>  {
                     if (!!tasks && tasks.length > 0) {
                         this.taskPanelData.next(this.resolveTransitionTask(tasks));
+                    } else {
+                        const isLoading = async.transform(this.loading$);
```

Comment:
I don't believe the async pipe is ment to be used like this.

what do we want to achieve? who initializes the `loading$` attribute? Why is it an `Observable` and not a `LoadingEmitter`? Our `LoadingEmitter` implementation is a behaviour subject, and we can get its current loading value without tricks.

Reply:
According to `AsyncPipe` Angular documentation it automatically handles the subscribe and unsubscribe to subjects, and as of documentation, it can be used in code as well. If you check the whole file, you can see that the loading is from `TaskViewService` and is implemented this way since May 2020. I cannot use `LoadingEmitter` as the `TaskViewService` cannot return that one. If you wish, I can refactor it.

Reply:
if the documentation says that this is fine, then we can leave it. It just seems weird to inject the AsyncPipe class.